OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.kt

```diff
@@ -0,0 +1,886 @@
+                binding.loadingError.apply {
+                    root.setVisibility(true)
+                    dismiss.setVisibility(true)
+                    dismiss.setOnClickListener(onCloseClick())
+                    state.setState(EdxErrorState.State.LOAD_ERROR, Screen.COURSE_DASHBOARD)
+                    state.setActionListener(onCloseClick())
+                }
+                binding.root
+            } else {
+                // The case where we need to fetch course's data based on its courseId
+                fetchCourseById()
+                FrameLayout(requireActivity()).apply {
+                    layoutParams = FrameLayout.LayoutParams(
+                        FrameLayout.LayoutParams.MATCH_PARENT,
+                        FrameLayout.LayoutParams.MATCH_PARENT
+                    )
+                    addView(inflater.inflate(R.layout.loading_indicator, container, false).apply {
+                        setVisibility(true)
+                    })
+                }
+            }
+        }
+
+        binding = FragmentCourseTabsDashboardBinding.inflate(inflater, container, false)
+        courseData = arguments?.getSerializable(Router.EXTRA_COURSE_DATA) as EnrolledCoursesResponse
+
+        setHasOptionsMenu(courseData.course.coursewareAccess.hasAccess())
+
+        if (!courseData.course.coursewareAccess.hasAccess()) {
+            setupToolbar(false)
+            val auditAccessExpired = !courseData.auditAccessExpires.isNullOrEmpty() &&
+                    Date().after(DateUtil.convertToDate(courseData.auditAccessExpires))
+            if (auditAccessExpired) {
+                if (courseData.isUpgradeable && environment.appFeaturesPrefs.isValuePropEnabled()) {
+                    setupIAPLayout()
+                } else {
+                    binding.accessError.let { error ->
+                        error.setVisibility(true)
+                        error.setState(State.AUDIT_ACCESS_EXPIRED, null)
+                        error.setPrimaryButtonListener(onFindCourseClick())
+                    }
+                }
+            } else if (!courseData.course.isStarted) {
+                binding.accessError.let { error ->
+                    error.setVisibility(true)
+                    error.setState(
+                        State.NOT_STARTED,
+                        DateUtil.formatCourseNotStartedDate(courseData.course.start)
+                    )
+                    error.setPrimaryButtonListener(onFindCourseClick())
+                }
+            } else {
+                //Todo Remove when Next Session Enrollment feature is added
+                val errorLayoutBinding =
+                    FragmentDashboardErrorLayoutBinding.inflate(inflater, container, false)
+                errorLayoutBinding.errorMsg.setText(R.string.course_not_started)
+                return errorLayoutBinding.root
+            }
+        } else {
+            setupToolbar(true)
```

Comment:
better to move this statement inside `setViewPager()`

Reply:
Any particular reason? IMO, it's good that we have it separated.

Reply:
we are getting data of tab_names from this method so better to set it inside this. 
in other case `setupToolbar(false)` we don't need tabs so we can call it separately.